include/s3select.h

```diff
@@ -2347,6 +2393,11 @@ class csv_object : public base_s3object
     do
     {
 
+      if(m_is_limit_on && m_processed_rows >= m_limit)
+      {
+        limit_reached = true;
+        break;
+      }
```

Comment:
it's not the correct location, 
this method receives chunks of data (which may contain many rows)

as commented, the limit operator should be implemented on `base_s3object`
per all data sources.